lib/src/components/text_input/text_input.dart

```diff
@@ -0,0 +1,446 @@
+  }
+}
+
+/// The current state of a [ZetaTextInput]
+class ZetaTextInputState extends State<ZetaTextInput> implements ZetaFormFieldState {
+  late final TextEditingController _controller;
+  final GlobalKey<FormFieldState<String>> _key = GlobalKey();
+  ZetaColors get _colors => Zeta.of(context).colors;
+
+  bool _hovered = false;
+
+  String? _errorText;
+
+  Color get _backgroundColor {
+    if (widget.disabled) {
+      return _colors.surfaceDisabled;
+    }
+    if (_errorText != null) {
+      return _colors.error.shade10;
+    }
+    return _colors.surfacePrimary;
+  }
+
+  TextStyle get _baseTextStyle {
+    TextStyle style = ZetaTextStyles.bodyMedium;
+    if (widget.size == ZetaWidgetSize.small) {
+      style = ZetaTextStyles.bodyXSmall;
+    }
+    return style;
+  }
+
+  EdgeInsets get _contentPadding {
+    switch (widget.size) {
+      case ZetaWidgetSize.large:
+        return const EdgeInsets.symmetric(horizontal: ZetaSpacing.x3, vertical: ZetaSpacing.x4);
+      case ZetaWidgetSize.small:
+      case ZetaWidgetSize.medium:
+        return const EdgeInsets.symmetric(horizontal: ZetaSpacing.x3, vertical: ZetaSpacing.x3);
+    }
+  }
+
+  TextStyle get _affixStyle {
+    Color color = _colors.textSubtle;
+    if (widget.disabled) {
+      color = _colors.textDisabled;
+    }
+    return _baseTextStyle.copyWith(color: color);
+  }
+
+  BoxConstraints get _affixConstraints {
+    late final double size;
+    switch (widget.size) {
+      case ZetaWidgetSize.large:
+        size = ZetaSpacing.x10;
+      case ZetaWidgetSize.medium:
+        size = ZetaSpacing.x8;
+      case ZetaWidgetSize.small:
+        size = ZetaSpacing.x6;
+    }
+    return BoxConstraints(
+      minWidth: size,
+    );
+  }
+
+  Widget? get _prefix {
+    if (widget.prefix != null) return widget.prefix;
+    if (widget.prefixText != null) {
+      final style = widget.prefixTextStyle ?? _affixStyle;
+      return Center(
+        widthFactor: 0,
+        child: Text(
+          widget.prefixText!,
+          style: style,
+        ),
+      ).paddingStart(ZetaSpacing.x2);
+    }
+
+    return null;
+  }
+
+  Widget? get _suffix {
+    if (widget.suffix != null) return widget.suffix;
+    if (widget.suffixText != null) {
+      final style = widget.suffixTextStyle ?? _affixStyle;
+      return Center(
+        widthFactor: 0,
+        child: Text(
+          widget.suffixText!,
+          style: style,
+        ),
+      ).paddingEnd(ZetaSpacing.x2);
+    }
+
+    return null;
+  }
+
+  OutlineInputBorder get _baseBorder => OutlineInputBorder(
+        borderRadius: widget.rounded ? ZetaRadius.minimal : ZetaRadius.none,
+        borderSide: BorderSide(color: _hovered ? _colors.borderSelected : _colors.borderSubtle),
+      );
+
+  OutlineInputBorder get _focusedBorder => _baseBorder.copyWith(
+        borderSide: BorderSide(color: _colors.primary.shade50, width: ZetaSpacing.x0_5),
+      ); // TODO(mikecoomber): change to colors.borderPrimary when added
```

Comment:
todo

Reply:
this color isn't in Zeta yet, hence the todo